### PR TITLE
teleport-cluster chart: set strategy to 'Recreate' if chartMode is 'standalone' (fixes #11484)

### LINKED ISSUE #11484: Helm chart teleport-cluster: "Multi-Attach error for volume" when upgrading via "helm upgrade"
### Description

I am running the teleport-cluster helm chart in my Kubernetes cluster.

Here is the values.yaml:
```
clusterName: teleport.<redacted>
kubeClusterName: nucs.<redacted>

standalone:
  volumeSize: 1Gi
service:
  type: ClusterIP
highAvailability:
  certManager:
    enabled: true
    issuerName: letsencrypt-production
    issuerKind: ClusterIssuer

podSecurityPolicy:
  enabled: false
```
**What happened**:

When upgrading via ` helm upgrade teleport-cluster teleport/teleport-cluster --namespace=teleport-cluster -f Teleport_Cluster/values.yaml` I get a second replicaset in the namespace, so I have one for the previous/running version and one for the one that I want to upgrade to.

However the pod for the new replicaset never starts, as it cannot attach to the volume:

```
Multi-Attach error for volume "pvc-2f570303-c1ee-43bb-b847-06686095b078" Volume is already used by pod(s) teleport-cluster-8487459d6c-gs2hb
  Warning  FailedMount             2m13s  kubelet                  Unable to attach or mount volumes: unmounted volumes=[data], unattached volumes=[data kube-api-access-spk8f teleport-tls config]: timed out waiting for the condition
```

I can manually scale the old replicaSet to only 1 replica, then the volume can get attached and the new pod starts running and everything is fine.

**What you expected to happen**:

Ideally the upgrade would succeed without manual interaction.

**Possible solution**:

The default strategy for a deployment is `RollingUpdate`, if it is not set. It might be a good idea to use the `Recreate` strategy, as a rolling Upgrade is not possible due to the volume being only attachable to one pod.
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Install teleport-cluster via helm chart in older version
2. wait for everything to run
3. upgrade to latest version using `helm upgrade ...`

### Server Details
- Teleport version (run `teleport version`): 8.3.4 => 9.0.0 => 9.0.1 => 9.0.2
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): Three-node K3s cluster on baremetal

### PR DESCRIPTION
**Pull request by johanneskastl**
Fixes #11484 

In standalone mode a PVC is created. The volume cannot be attached to more than one pod (depending on the storage available), so using a `Recreate` strategy is needed to avoid stuck deployments on upgrades.

### CODE REVIEW

**Comment by webvictim** [comment]
Hey @johanneskastl - thanks for the PR. Please take a look at https://github.com/gravitational/teleport/blob/master/examples/chart/CONTRIBUTING.md for some additional testing we'd need.

**Comment by johanneskastl** [comment]
> Hey @johanneskastl - thanks for the PR. Please take a look at https://github.com/gravitational/teleport/blob/master/examples/chart/CONTRIBUTING.md for some additional testing we'd need.

Thanks for the hint @webvictim! I have added a test to `tests/deployment_test.yaml` that verifies that it sets the strategy in standalone mode. Do you recommend adding more tests?

Did I miss any other steps?

File: examples/chart/teleport-cluster/templates/deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ spec:
   {{- else }}
   replicas: 1
   {{- end }}
+  {{- if eq .Values.chartMode "standalone" }}
+  strategy:
+    type: 'Recreate'
```
Comment (webvictim) [inline examples/chart/teleport-cluster/templates/deployment.yaml:26]:
```suggestion
    type: Recreate
```

**Review by webvictim — COMMENTED** [review]
@johanneskastl Thanks! Looks great!

Just one final ask: could you also please add three extra tests to make sure that the strategy is not set (so will default to `RollingUpdate`) for the other modes (`aws`, `gcp` and `custom`)? 

Something similar to this section: https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-cluster/tests/deployment_test.yaml#L234-L310 - you should be able to use the same values files.

File: examples/chart/teleport-cluster/templates/deployment.yaml (hunk 1 above)
Comment (johanneskastl) [reply examples/chart/teleport-cluster/templates/deployment.yaml]: Fixed.

**Comment by johanneskastl** [comment]
Hey @webvictim, I added three more tests for AWS/GCP/Custom mode to not contain the `strategy` setting. Could you have a look if this is fine? Anything else missing?

**Review by dirien — APPROVED** [review]
Hi @johanneskastl,

that looks good for me! Looking forward for this PR as it blocks us too currently.

**Comment by zmb3** [comment]
Looks good to me @johanneskastl - we just need to commit the `examples/chart/teleport-cluster/tests/__snapshot__/deployment_test.yaml.snap` and we should be good to merge this.

Thanks for the contribution!

**Review by webvictim — COMMENTED** [review]
LGTM. As @zmb3 says, please commit the snapshot and we'll be good to go!

**Comment by johanneskastl** [comment]
@webvictim  @zmb3 I commited the file, please have a look.

**Review by zmb3 — APPROVED** [review]

**Comment by johanneskastl** [comment]
Rebased on current master and solved conflict in `examples/chart/teleport-cluster/tests/__snapshot__/deployment_test.yaml.snap`. 

Can this be merged? Or is something missing?

**Comment by johanneskastl** [comment]
@webvictim I'll stop rebasing this, as I cannot run the tests anyway.

Please ping me if I can assists or if fixes are needed.

**Comment by webvictim** [comment]
@johanneskastl Thanks - I'm sure it's fine, we'll take a look when people are back at work on Monday.

**Review by webvictim — APPROVED** [review]

**Comment by webvictim** [comment]
It looks like the review bot is refusing to approve this because reasons. I'll open a new buddy PR and get it merged.

**Comment by johanneskastl** [comment]
> It looks like the review bot is refusing to approve this because reasons. I'll open a new buddy PR and get it merged.

Thanks @webvictim

**Comment by johanneskastl** [comment]
@webvictim I see you merged 11693 containing my changes, so can we close this one?

**Comment by webvictim** [comment]
@johanneskastl Thanks for the contribution!

**Pull request closed without merging**